pandas/tests/io/test_excel.py

```diff
@@ -1168,6 +1212,18 @@ def test_read_xlrd_book(self, ext):
                                 engine=engine, index_col=0)
             tm.assert_frame_equal(df, result)
 
+@td.skip_if_no("openpyxl")
+@pytest.mark.parametrize("ext", ['.xlsx', '.xlsm'])
+class TestOpenpyxlReader(ReadingTestsBase):
```

Comment:
Instead of this subclass you should just be able to add `openpyxl` to the parametrization on the `set_engine` method of the superclass. Might need to work around a lack of support for `.xls` files though

Reply:
Don't agree with you there, please the previous discussion we had on this above. This would require revising the general structure of the read tests (which is fundamentally different from the write tests, so that may be a good idea anyway)